Core/include/Acts/Surfaces/detail/DiscSurface.ipp

```diff
@@ -19,58 +19,54 @@ inline Vector2D DiscSurface::localCartesianToPolar(
                   atan2(lcart[eBoundLoc1], lcart[eBoundLoc0]));
 }
 
-inline void DiscSurface::initJacobianToGlobal(const GeometryContext& gctx,
-                                              BoundToFreeMatrix& jacobian,
-                                              const Vector3D& position,
-                                              const Vector3D& direction,
-                                              const BoundVector& pars) const {
-  // The trigonometry required to convert the direction to spherical
-  // coordinates and then compute the sines and cosines again can be
-  // surprisingly expensive from a performance point of view.
-  //
-  // Here, we can avoid it because the direction is by definition a unit
-  // vector, with the following coordinate conversions...
-  const double x = direction(0);  // == cos(phi) * sin(theta)
-  const double y = direction(1);  // == sin(phi) * sin(theta)
-  const double z = direction(2);  // == cos(theta)
-
-  // ...which we can invert to directly get the sines and cosines:
-  const double cos_theta = z;
-  const double sin_theta = sqrt(x * x + y * y);
-  const double inv_sin_theta = 1. / sin_theta;
-  const double cos_phi = x * inv_sin_theta;
-  const double sin_phi = y * inv_sin_theta;
-  // retrieve the reference frame
-  const auto rframe = referenceFrame(gctx, position, direction);
-
+inline BoundToFreeMatrix DiscSurface::jacobianLocalToGlobal(
+    const GeometryContext& gctx, const BoundVector& boundParams) const {
+  // Convert angles to global unit direction vector
+  const Vector3D direction = makeDirectionUnitFromPhiTheta(
+      boundParams[eBoundPhi], boundParams[eBoundTheta]);
+  // Convert local position to global position vector
+  const Vector2D local(boundParams[eBoundLoc0], boundParams[eBoundLoc1]);
+  const Vector3D position = localToGlobal(gctx, local, direction);
```

Comment:
The direction vector is only needed for the `localToGlobal` call, right? Since here we know that we are on a disc surface and thus the transformation does not require the direction, could we do the transformation manually w/o constructing the direction vector?

Reply:
The `direction` is also a parameter of the `referenceFrame` call (though it's not used for the `DiscSurface`). Now, this transform is done through the `transformBoundToFreeParameters`. Since it returns the full free parameters, i.e. the `direction` is constructed anyway, so we could just use it.